SimDataFormats/CaloAnalysis/interface/SimCluster.h

```diff
@@ -186,12 +192,27 @@ class SimCluster {
     return result;
   }
 
+  /** @brief Returns list of rechit IDs and energies for this SimCluster */
+  std::vector<std::pair<uint32_t, float>> hits_and_energies() const {
+    std::vector<std::pair<uint32_t, float>> result;
+    for (size_t i = 0; i < hits_.size(); ++i) {
+      result.emplace_back(hits_[i], energies_[i]);
+    }
+    return result;
+  }
+
   /** @brief clear the hits and fractions list */
   void clearHitsAndFractions() {
     std::vector<uint32_t>().swap(hits_);
     std::vector<float>().swap(fractions_);
   }
 
+  /** @brief clear the hits and energies list */
+  void clearHitsAndEnergies() {
```

Comment:
Should this function be called in `CaloTruthAccumulator`? I'm just wondering the use case for a partial copy of `clearHitsAndFractions()` function, that is not called anywhere in this PR.